force-app/main/default/classes/NKS_KnowledgeBrokenLinksFinder.cls

```diff
@@ -0,0 +1,245 @@
+                article.AdvisorInformation__c,
+                article.AdvisorInformationInternal__c,
+                article.Article__c,
+                article.EmployerInformation__c,
+                article.EmployerInformationInternal__c,
+                article.NKS_English__c,
+                article.InternationalInformation__c,
+                article.InternationalInformationInternal__c,
+                article.NKS_Legislation__c,
+                article.NKS_Nav_no__c,
+                article.NKS_Nynorsk__c,
+                article.NKS_Resources__c,
+                article.NKS_Summary__c,
+                article.NKS_UserInformation__c,
+                article.WhoDoesWhat__c
+            };
+
+            if (strList.size() > 0) {
+                // find all links
+                for (String str : strList) {
+                    try {
+                        Matcher matcher = linksPattern.matcher(str);
+                        while (matcher.find()) {
+                            String groupValue = matcher.group();
+                            matches.add(groupValue);
+                        }
+                    } catch (Exception ex) {
+                        //System.debug(ex.getMessage());
+                    }
+                }
+            }
+            //System.debug('Matches: ' + matches);
+
+            /**
+             * Endpoints defined as named credentials
+             */
+            if (matches.size() > 0) {
+                for (String str : matches) {
+                    HttpRequest req = new HttpRequest();
+
+                    if (Test.isRunningTest()) {
+                        System.debug('Running tests.');
+                    } else {
+                        Integer statusCode = 0;
+                        for (String link : links) {
+                            if (str.contains(link)) {
+                                try {
+                                    HttpResponse res = calloutHttpReq(
+                                        req,
+                                        'callout:' +
+                                        link.replace('.', '_') +
+                                        str.substringAfter(link)
+                                    );
+                                    if (link.contains('.no')) {
+                                        statusCode = detectRedirect(req, res);
+                                    } else {
+                                        statusCode = res.getStatusCode();
+                                    }
+
+                                    if (statusCode != 200 && statusCode != 500) {
+                                        resultList.add(str);
+                                    }
+                                } catch (Exception ex) {
+                                    //System.debug('It is possbile this link is broken, check the link.');
+                                }
```

Comment:
Kan vi ikke breake den andre for loopen etter vi har funnet rett link?

Reply:
Nei, da vil loopen stoppe å finne andre lenker :)